core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/AbstractForEachDataStream.java

```diff
@@ -34,6 +38,10 @@ public final void buildNode(DataNodeBuildHelper<Solution_> buildHelper) {
         TupleLifecycle<UniTuple<A>> tupleLifecycle = buildHelper.getAggregatedTupleLifecycle(childStreamList);
         var outputStoreSize = buildHelper.extractTupleStoreSize(this);
         var node = getNode(tupleLifecycle, outputStoreSize);
+        if (shouldIncludeNull && !(node instanceof ForEachFromSolutionUniNode<?, ?>)) {
```

Comment:
I recommend updating the `AbstractForEachUniNode` contract to include a method similar to `acceptNullValue()` instead of relying on `instanceof`.